bin/cli.rs

```diff
@@ -0,0 +1,126 @@
+#![allow(missing_docs)]
+use std::{
+    collections::BTreeMap,
+    fs::{create_dir_all, File},
+    io::BufReader,
+    num::NonZeroUsize,
+    thread,
+};
+
+use alloy_consensus::constants::KECCAK_EMPTY;
+use alloy_primitives::{Address, B256};
+use alloy_provider::{Provider, ProviderBuilder};
+use alloy_rpc_types::{BlockId, BlockTransactionsKind};
+use clap::{Parser, Subcommand};
+use pevm::{
+    chain::{PevmChain, PevmEthereum},
+    EvmAccount, EvmCode, Pevm, RpcStorage,
+};
+use reqwest::Url;
+use tokio::runtime;
+
+#[derive(Parser, Debug)]
+pub struct Cli {
+    #[clap(subcommand)]
+    pub subcmd: Commands,
+}
+
+#[derive(Subcommand, Debug)]
+pub enum Commands {
+    /// Snapshot a block to disk.
+    Snapshot {
+        /// The ID of the block.
+        block_id: BlockId,
+        /// The RPC url to connect to.
+        rpc_url: String,
+        /// The output directory to write the snapshot to.
+        output_dir: String,
+        /// Path to bytecodes.
+        bytecode_path: Option<String>,
+    },
+}
+
+pub fn main() {
```

Comment:
It'd be nice if we could write this `async` with Tokio, or let's leave a TODO for it.

Reply:
Left a todo. The stack is panicking with:

```
Cannot start a runtime from within a runtime. This happens because a function (like `block_on`) attempted to block the current thread while the thread is being used to drive asynchronous tasks.
```

I'm assuming this has something to do with the pevm execution stack somewhere. Can look to address in a follow up PR.

Reply:
Yeah, we must also make `RpcStorage` functions `async`. Other than that the main pevm execution stack doesn't depend on Tokio!

A future PR is fine, thanks.